kusto/data/value/value.go

```diff
@@ -534,12 +536,15 @@ func (t Timespan) Marshal() string {
 	seconds := val / time.Second
 	val = val - (seconds * time.Second)
 	switch {
-	case minutes < 10:
+	case seconds < 10:
 		sb.WriteString(fmt.Sprintf("0%d", int(seconds)))
```

Comment:
You can use %02d which will pad the printed value with zeroes.
This can simply most of this function and especially the fractional part

Reply:
Change and code simplified.  Completely forgot about this, good catch.